homeassistant/components/telnet/switch.py

```diff
@@ -96,82 +98,71 @@ class TelnetSwitch(SwitchEntity):
 
     def __init__(
         self,
-        hass,
-        object_id,
-        resource,
-        port,
-        friendly_name,
-        command_on,
-        command_off,
-        command_state,
-        value_template,
-        timeout,
-    ):
+        hass: HomeAssistant,
+        object_id: str,
+        resource: str,
+        port: int,
+        friendly_name: str,
+        command_on: str,
+        command_off: str,
+        command_state: str,
+        value_template: Template,
+        timeout: float,
+    ) -> None:
         """Initialize the switch."""
         self._hass = hass
         self.entity_id = ENTITY_ID_FORMAT.format(object_id)
         self._resource = resource
         self._port = port
-        self._name = friendly_name
-        self._state = False
+        self._attr_name = friendly_name
+        self._state: bool = False
         self._command_on = command_on
         self._command_off = command_off
         self._command_state = command_state
         self._value_template = value_template
         self._timeout = timeout
+        self._attr_should_poll = bool(command_state)
 
-    def _telnet_command(self, command):
+    def _telnet_command(self, command: str) -> str | None:
         try:
             telnet = telnetlib.Telnet(self._resource, self._port)
             telnet.write(command.encode("ASCII") + b"\r")
             response = telnet.read_until(b"\r", timeout=self._timeout)
-            _LOGGER.debug("telnet response: %s", response.decode("ASCII").strip())
-            return response.decode("ASCII").strip()
         except OSError as error:
             _LOGGER.error(
                 'Command "%s" failed with exception: %s', command, repr(error)
             )
             return None
+        _LOGGER.debug("telnet response: %s", response.decode("ASCII").strip())
+        return response.decode("ASCII").strip()
 
     @property
-    def name(self):
-        """Return the name of the switch."""
-        return self._name
-
-    @property
-    def should_poll(self):
-        """Only poll if we have state command."""
-        return self._command_state is not None
-
-    @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if device is on."""
         return self._state
 
-    @property
-    def assumed_state(self):
```

Comment:
Why do we remove this?

Reply:
It's not necessary to have so removed it to trim down the code but perhaps better to keep for clarity.

Reply:
Removing it will change behavior. I don't think we can say it's not needed without further explanation.

Reply:
I did also change the things consuming the info but I'll revert this anyhow to make the code clearer.